scala/tempo/src/test/scala/com/databrickslabs/tempo/SMATests.scala

```diff
@@ -0,0 +1,46 @@
+package com.databrickslabs.tempo
+
+import com.databrickslabs.tempo.ml.stat.SMA
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.{DoubleType, StringType}
+import org.scalatest.FunSpec
+
+class SMATests
+	extends FunSpec with SparkSessionTestWrapper
+{
+	val schema = buildSchema(Seq(("symbol", StringType),
+	                             ("event_ts", StringType),
+	                             ("trade_pr", DoubleType)))
+
+	val expectedSchema = buildSchema(Seq(("symbol", StringType),
+	                                     ("event_ts", StringType),
+	                                     ("trade_pr", DoubleType),
+	                                     ("sma_trade_pr", DoubleType)))
+
+	lazy val df = buildTestDF(schema,
+	                          Seq(Row("S1", "2020-08-01 00:00:10", 8.0),
+	                              Row("S1", "2020-08-01 00:01:12", 4.0),
+	                              Row("S1", "2020-08-01 00:02:23", 2.0),
+	                              Row("S2", "2020-09-01 00:02:10", 8.0),
+	                              Row("S2", "2020-09-01 00:19:12", 16.0),
+	                              Row("S2", "2020-09-01 00:19:12", 32.0)),
+	                          "event_ts")
+
+	lazy val dfExpected = buildTestDF(expectedSchema,
+	                                  Seq(Row("S1", "2020-08-01 00:00:10", 8.0, 8.0),
+	                                      Row("S1", "2020-08-01 00:01:12", 4.0, 6.0),
+	                                      Row("S1", "2020-08-01 00:02:23", 2.0, 4.0 + (2.0/3.0)),
+	                                      Row("S2", "2020-09-01 00:02:10", 8.0, 8.0),
+	                                      Row("S2", "2020-09-01 00:19:12", 16.0, 12.0),
+	                                      Row("S2", "2020-09-01 00:19:12", 32.0, 18.0 + (2.0/3.0))),
+	                                  "event_ts" )
+
+	it("SMA Transformer"){
+		val tsdf = TSDF(df, tsColumnName = "event_ts", partitionColumnNames = Seq("symbol"))
+
+		val sma_tx = SMA("trade_pr","ema_trade_pr",2)
+		val smaDf = sma_tx.transform(tsdf).df
```

Comment:
Can we add a test with repeated transforms so we emulate something complex but whats done in practice? Something like: 

pipeline = [SMA("trade_pr", "sma_trade_pr_20", windowLength=20), SMA("trade_pr", "sma_trade_pr_50", 50), EMA(...)]

Reply:
good idea - I can put something like that together

Reply:
currently blocked by issue [Tempo-10](https://databricks.atlassian.net/browse/TEMPO-10)